crates/resolver/src/config.rs

```diff
@@ -952,6 +952,12 @@ pub struct ResolverOpts {
     pub preserve_intermediates: bool,
     /// Try queries over TCP if they fail over UDP.
     pub try_tcp_on_error: bool,
+    /// Try queries over TCP for I/O errors only.
```

Comment:
I don't think this should be yet another configuration option and I think it might be wrong? AIUI there are non-I/O failure modes that might be resolved by communicating over TCP (for example if a response does not fit in a UDP datagram).

Reply:
I see this more as feature flag, so the default behavior remains unchanged and a user can "opt-in" into this new one.

Now, my general problem is that we have valid responses that are converted into "errors" and then retried when  `try_tcp_on_error` is on: in fact, we have certain cases of ["NoError" converted into a error](https://github.com/hickory-dns/hickory-dns/blob/v0.25.0-alpha.1/crates/proto/src/error.rs#L436-L458) (this makes no sense). I'm open to suggestions on how to get out of this, but retrying these cases over TCP and expecting a different response is not sane.

As for responses that don't fit in a UDP packet, they should have the `TR` flag set and retry [should](https://github.com/hickory-dns/hickory-dns/blob/v0.25.0-alpha.1/crates/resolver/src/name_server/name_server_pool.rs#L263) happen. Perhaps there is a different use-case?

Reply:
Making this opt-in seems like a cop out to me, we should figure out the correct behavior and implement it and it doesn't feel like there's an obvious reason that should depend on user configuration (which is pretty hard to discover). Let's wait for some insights from @bluejekyll...

Reply:
I'll respond in line, but I agree with @djc, that we don't want a new value here. I thought I fixed this recently, maybe I got it wrong?